### PR TITLE
[Fix] Fix the problem that the instance cannot reuse when overwriting the file client.

### PR DESCRIPTION
**Pull request by 274869388**
## Motivation

The motivation of this PR is to fix the problem that the instance cannot reuse when overwriting the file client.

## Modification

1. FileClient
    - Remove _overridden_backends and _overridden_prefixes, because these variables will not used.
    - Pop all instances of `StorageBackend` when it is overwrite.

## BC-breaking (Optional)

No

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/fileio/file_client.py [hunk 1]
```diff
@@ -912,7 +910,16 @@ def _register_backend(cls, name, backend, force=False, prefixes=None):
...
+                    clear_backend = cls._prefix_to_backends[prefix]
+                    if isinstance(clear_backend, list):
+                        clear_backend = tuple(clear_backend)
+                    clear_key = [
+                        arg_key
+                        for arg_key, instance in cls._instances.items()
+                        if isinstance(instance.client, clear_backend)
+                    ]
+                    for key in clear_key:
+                        cls._instances.pop(key)
```
Comment (zhouzaida) [inline mmcv/fileio/file_client.py:922]:
```suggestion
                    for arg_key, instance in list(cls._instances.items()):
                        if isinstance(instance.client, clear_backend):
                            cls._instances.pop(key)
```

File: mmcv/fileio/file_client.py [hunk 2]
```diff
@@ -900,7 +893,12 @@ def _register_backend(cls, name, backend, force=False, prefixes=None):
                 'add "force=True" if you want to override it')
 
         if name in cls._backends and force:
-            cls._overridden_backends.add(name)
+            clear_key = [
+                arg_key for arg_key, instance in cls._instances.items()
+                if isinstance(instance.client, cls._backends[name])
+            ]
+            for key in clear_key:
+                cls._instances.pop(key)
```
Comment (zhouzaida) [inline mmcv/fileio/file_client.py:901]:
```suggestion
            for arg_key, instance in list(cls._instances.items()):
                if isinstance(instance.client, cls._backends[name]):
                    cls._instances.pop(key)
```

**Comment by zhouzaida** [comment]
Please refine the comment in https://github.com/open-mmlab/mmcv/pull/1747/files#diff-61896dd6d4f845655707a5927f37e1906ef6972ca77d48d634a27e305fe9cccaL645

File: mmcv/fileio/file_client.py (hunk 1 above)
Comment (274869388) [reply mmcv/fileio/file_client.py]: done

**Review by zhouzaida — APPROVED** [review]

File: mmcv/fileio/file_client.py [hunk 3]
```diff
@@ -804,9 +799,7 @@ def __new__(cls, backend=None, prefix=None, **kwargs):
             arg_key += f':{key}:{value}'
 
         # if a backend was overridden, it will create a new object
```
Comment (ZwwWayne) [inline mmcv/fileio/file_client.py:801]: is this comment still correct?

Comment (274869388) [reply mmcv/fileio/file_client.py:801]: This comment should be deleted.

Comment (274869388) [reply mmcv/fileio/file_client.py]: done

**Pull request merged**